src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +44,30 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter FORMAT = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate dateFromLocal = LocalDate.parse(dateFrom, FORMAT);
+        LocalDate dateToLocal = LocalDate.parse(dateTo, FORMAT);
+        if (!dateFromLocal.isBefore(dateToLocal)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuffer result = new StringBuffer("Отчёт за период " + dateFrom + " - " + dateTo);
+        int salary = 0;
+        for (String retVal: names) {
+            salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String [] oneString = data[i].split(" ");
+                LocalDate dateOnThisLine = LocalDate.parse(oneString[0], FORMAT);
+                if (oneString[1].equals(retVal) == true
+                        && dateOnThisLine.isBefore(dateFromLocal) == false
+                        && dateOnThisLine.isAfter(dateToLocal) == false) {
```

Comment:
same

Reply:
i changed by isBefore and isAfter without FALSE or TRUE